pkg/predicate/predicate_test.go

```diff
@@ -94,182 +100,185 @@ var _ = Describe("Druid Predicate", func() {
-					gomega.Expect(predicate.Update(updateEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Delete(deleteEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Generic(genericEvent)).To(gomega.BeTrue())
-				})
+			It("should return false", func() {
+				gomega.Expect(pred.Create(createEvent)).To(gomega.BeFalse())
+				gomega.Expect(pred.Update(updateEvent)).To(gomega.BeFalse())
+				gomega.Expect(pred.Delete(deleteEvent)).To(gomega.BeTrue())
+				gomega.Expect(pred.Generic(genericEvent)).To(gomega.BeFalse())
 			})
 		})
 
-		Describe("#HasOperationAnnotation", func() {
-			Context("when has no operation annotation", func() {
-				BeforeEach(func() {
-					obj = &druidv1alpha1.Etcd{
-						ObjectMeta: metav1.ObjectMeta{
-							Annotations: make(map[string]string),
+		Context("when has operation annotation", func() {
+			BeforeEach(func() {
+				obj = &druidv1alpha1.Etcd{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: map[string]string{
+							v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
 						},
-					}
-				})
-
-				It("should return false", func() {
-					predicate := HasOperationAnnotation()
+					},
+				}
+			})
 
-					gomega.Expect(predicate.Create(createEvent)).To(gomega.BeFalse())
-					gomega.Expect(predicate.Update(updateEvent)).To(gomega.BeFalse())
-					gomega.Expect(predicate.Delete(deleteEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Generic(genericEvent)).To(gomega.BeFalse())
-				})
+			It("should return true", func() {
+				gomega.Expect(pred.Create(createEvent)).To(gomega.BeTrue())
+				gomega.Expect(pred.Update(updateEvent)).To(gomega.BeTrue())
+				gomega.Expect(pred.Delete(deleteEvent)).To(gomega.BeTrue())
+				gomega.Expect(pred.Generic(genericEvent)).To(gomega.BeTrue())
 			})
+		})
+	})
 
-			Context("when has operation annotation", func() {
-				BeforeEach(func() {
-					obj = &druidv1alpha1.Etcd{
-						ObjectMeta: metav1.ObjectMeta{
-							Annotations: map[string]string{
-								v1beta1constants.GardenerOperation: v1beta1constants.GardenerOperationReconcile,
-							},
-						},
-					}
-				})
+	Describe("#OR", func() {
+		var pred predicate.Predicate
 
-				It("should return false", func() {
-					predicate := HasOperationAnnotation()
+		JustBeforeEach(func() {
+			pred = predicate.Or(
+				HasOperationAnnotation(),
+				LastOperationNotSuccessful(),
+			)
+		})
 
-					gomega.Expect(predicate.Create(createEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Update(updateEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Delete(deleteEvent)).To(gomega.BeTrue())
-					gomega.Expect(predicate.Generic(genericEvent)).To(gomega.BeTrue())
-				})
+		Context("when has neither operation annotation nor last error", func() {
+			BeforeEach(func() {
+				obj = &druidv1alpha1.Etcd{
+					ObjectMeta: metav1.ObjectMeta{
+						Annotations: make(map[string]string),
+					},
+				}
+			})
+
+			It("should return true", func() {
```

Comment:
@amshuman-kr Thanks for making the changes and the rebase too! Just one small typo here to fix, otherwise LGTM

Reply:
😅